### PR TITLE
Minor refactor of cached packets (and storage) fetch

### PR DESCRIPTION
**Pull request by trvrnrth**
I was looking at this again whilst testing and it seemed like this was perhaps easier to follow.

Also simplifies storage data load as the wrapper function doesn't really do much.

### CODE REVIEW

File: custom_components/ramses_cc/broker.py [hunk 1]
```diff
@@ -92,15 +92,23 @@ async def start(self) -> None:
...
         )
 
-        cached_packets = (
-            self._filter_cached_packets(
-                client_state.get(SZ_PACKETS, {}), restore_schema
-            )
-            if restore_state
-            else {}
-        )
+        @callback
```
Comment (zxdavb) [inline custom_components/ramses_cc/broker.py:95]: Do we need this decorator?

I never 100% understood what it's for, but as its only ever going to be synchronously invoked by our `start()` (and nor, say, `ceate_task()` / `call_soon()`), I don't believe it's needed.

File: custom_components/ramses_cc/broker.py [hunk 2]
```diff
@@ -92,15 +92,23 @@ async def start(self) -> None:
...
+        def cached_packets() -> dict:
```
Comment (zxdavb) [inline custom_components/ramses_cc/broker.py:96]: Let's try not to use generic types if possible. At least `dict[str, Any]`, but we can have:
```python
def cached_packets() -> dict[str, str]:  # dtm_str, packet_as_str
```

Eventually, I'd be making everything strict-typed, and I've discovered it's a real pain to retro-fit... But I pick up so many bugs doing it.

**Review by zxdavb — CHANGES_REQUESTED** [review]

File: custom_components/ramses_cc/broker.py (hunk 2 above)
Comment (trvrnrth) [reply custom_components/ramses_cc/broker.py:96]: Yeah, good shout.

File: custom_components/ramses_cc/broker.py (hunk 1 above)
Comment (trvrnrth) [reply custom_components/ramses_cc/broker.py]: No, fair point. Hass will never be calling this directly so there should be no need.

**Pull request merged**